infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -108,6 +117,14 @@ public void init() throws TestGridInfrastructureException {
     @Override
     public InfrastructureProvisionResult provision(InfrastructureConfig infrastructureConfig, String infraRepoDir)
             throws TestGridInfrastructureException {
+        try {
+            infrastructureConfig = addAMIParameter(infrastructureConfig);
```

Comment:
AMI parameter should be added only if CF scripts requested it.. 

See how this is done in here: https://github.com/pasindujw/testgrid/blob/9b391d1e26ab8c82cc5d44653a2244fd15055842/infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java#L269